pkg/registry/apis/provisioning/repository/github.go

```diff
@@ -25,12 +25,12 @@ import (
 )
 
 type githubRepository struct {
-	logger   *slog.Logger
-	config   *provisioning.Repository
-	gh       pgh.Client
-	baseURL  *url.URL
-	linter   lint.Linter
-	renderer PreviewRenderer
+	logger        *slog.Logger
+	config        *provisioning.Repository
+	gh            pgh.Client
+	baseURL       *url.URL
+	linterFactory lint.LinterFactory
```

Comment:
Maybe a different issue... should githubRepository be using the parser to validate?  This would give it access to the same logic we can use everywhere that also checks what happens with dryRun, etc